cpp/velox/compute/WholeStageResultIterator.cc

```diff
@@ -118,46 +118,64 @@ std::shared_ptr<ColumnarBatch> WholeStageResultIterator::next() {
   return std::make_shared<VeloxColumnarBatch>(vector);
 }
 
+namespace {
+class ConditionalSuspendedSection {
+ public:
+  ConditionalSuspendedSection(velox::exec::Driver* driver, bool condition) {
+    if (condition) {
+      section_ = new velox::exec::SuspendedSection(driver);
+    }
+  }
+
+  virtual ~ConditionalSuspendedSection() {
+    if (section_) {
+      delete section_;
+    }
+  }
+
+  // singleton
+  ConditionalSuspendedSection(const ConditionalSuspendedSection&) = delete;
+  ConditionalSuspendedSection(ConditionalSuspendedSection&&) = delete;
+  ConditionalSuspendedSection& operator=(const ConditionalSuspendedSection&) = delete;
+  ConditionalSuspendedSection& operator=(ConditionalSuspendedSection&&) = delete;
+
+ private:
+  velox::exec::SuspendedSection* section_ = nullptr;
```

Comment:
nit: why not use unique_ptr?

Reply:
I'll open to both. Not a big deal. :)

Reply:
unique_ptr is better to me, let's use modern cpp.

Reply:
> unique_ptr is better to me, let's use modern cpp.

Can you elaborate? I am not sure modern cpp recommends using smart pointers in this case. `unique_ptr` should indeed be used when **ownership-transferring** is going to happen. However in this case we just define a owning pointer inside the class.

Just to make sure we don't create new rules casually to our CPP coding convention. The convention should be followed once built.

Reply:
Please refer to https://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines

Reply:
https://www.oreilly.com/library/view/effective-modern-c/9781491908419/ch04.html

I'm not going to discuss which usage is the best and create more coding rules, current implementation is ok, but I prefer use unique_ptr to replace manually `delete`, let language provided ability help us.

You can keep this usage if you like, this not cause problem.